src/opinion/actions/get-public-opinions.action.ts

```diff
@@ -22,15 +23,16 @@ export const getPublicOpinions = async <T, K>(
       filters,
     );
 
-    switch (response.status) {
-      case 200:
-        return response.data;
+    return response.data;
+  } catch (error: SherlError | Error | any) {
+    switch ((error as SherlError).data?.status) {
       case 403:
         throw errorFactory.create(OpinionErr.FETCH_PUBLIC_OPINIONS_FORBIDDEN);
       default:
-        throw errorFactory.create(OpinionErr.FETCH_FAILED);
+        throw getSherlError(
+          error,
+          errorFactory.create(OpinionErr.FETCH_FAILED),
```

Comment:
```suggestion
          errorFactory.create(OpinionErr.FETCH_PUBLIC_OPINIONS_FAILED),
```
ou (mais ça impact l'autre erreur mais c'est plus raccord avec l'action)
```suggestion
          errorFactory.create(OpinionErr.GET_PUBLIC_OPINIONS_FAILED),
```